reviews/NFTTrader/testing.md

```diff
@@ -0,0 +1,48 @@
+# Testing
+
+
+## Summary
+
+- I could not get these tests to run properly - I'm not sure if there was additional config or project files that were removed for the audit. It would have helped if these had been included.
+
+### Recommendations
+- Looks like the test file starts out with a bunch of helper functions. It might be nice to label this or split them out.
+- In general the test code is repetitive and could definitely be refactored.
+- While the deployment tests seem comprehensive, there is no attempt at any other kind of testing. Unit tests or a more structured approach to test writing would make it much easier to see where coverage is lacking.
```

Comment:
indeed true , explaning the previous review point .

Reply:
definitely agree. these tests as mentioned appear to be deployment tests/integration tests which although great don't serve to really have any unit tests to understand individual use-cases & failure points of the functions within each contract